src/AppInstallerCommonCore/ThreadGlobals.cpp

```diff
@@ -0,0 +1,55 @@
+#include "pch.h"
+#include "Public/ThreadGlobals.h"
+
+namespace AppInstaller::ThreadLocalStorage
+{
+    DiagnosticLogger& ThreadGlobals::GetDiagnosticLogger()
+    {
+        if (!m_pDiagnosticLogger)
+        {
+            try
+            {
+                std::call_once(diagLoggerInitOnceFlag, [this]()
+                {
+                    m_pDiagnosticLogger = std::make_unique<DiagnosticLogger>();
+                });
+            }
+            catch (std::exception& ex)
+            {
+                // May throw std::system_error if any condition prevents calls to call_once from executing as specified
+                // May throw std::bad_alloc or any exception thrown by the constructor of TelemetryTraceLogger
+                // Loggers are best effort and shouldn't block core functionality. So eat up the exceptions here
+                AICLI_LOG(Log, Error, << "Exception caught when creating DIAGNOSTIC Trace logger: " << ex.what() << '\n');
+            }
+        }
+        return *(m_pDiagnosticLogger.get());
+    }
+
+    TelemetryTraceLogger& ThreadGlobals::GetTelemetryLogger()
+    {
+        return *(m_pTelemetryLogger.get());
+    }
+
+    void ThreadGlobals::SetForCurrentThread()
+    {
+        try
+        {
+            m_pTelemetryLogger = std::make_unique<TelemetryTraceLogger>();
+        }
+        catch (std::exception& ex)
+        {
+            // May throw std::bad_alloc or any exception thrown by the constructor of TelemetryTraceLogger
+            // Loggers are best effort and shouldn't block core functionality. So eat up the exceptions here
+            AICLI_LOG(Log, Error, << "Exception caught when creating TELEMETRY Trace logger: " << ex.what() << '\n');

```

Comment:
Crashes as there is no global set yet.

Reply:
I am updating AICLI_LOG to only log to FileLogger for messages in Channel::Log.

Reply:
Never mind. My response would have worked only when TelemetryTraceLogger unique_ptr is made in GetTelemetryLogger() itself. This would definitely crash.